v2/tools/generator/internal/jsonast/swagger_type_extractor.go

```diff
@@ -569,19 +629,40 @@ func (extractor *SwaggerTypeExtractor) expandEnumsInPath(
 		}
 
 		enum, isEnum := paramType.(*astmodel.EnumType)
-		if isEnum {
-			toReplace := fmt.Sprintf("{%s}", resolved.Name)
-			var expanded []string
-
-			for _, path := range results {
-				for _, opt := range enum.Options() {
-					theOption := strings.Trim(opt.Value, "\"")
-					expanded = append(expanded, strings.ReplaceAll(path, toReplace, theOption))
-				}
-			}
+		if !isEnum {
+			// We only expand enums, skip everything else
+			continue
+		}
+
+		if len(enum.Options()) == 0 {
+			// This shouldn't happen but guard against it anyway
+			continue
+		}
+
+		// resource types come on the even counts: Microsoft.Group/resourceType/{resourceName}/{childResourceTypeParameter}/{childName}
```

Comment:
:thinking: ... "even counts" assumes you treat `Microsoft.Group` as item 0, but not if you start counting from 1.

Not sure how to clarify though ...

Reply:
When in doubt, give a better example:

```
		// resource types come on the even indices after the group.
		// Example: Microsoft.Group/resourceType/{resourceName}/{childResourceTypeParameter}/{childName},
		// has "resourceType" at index=0, "{childResourceTypeParameter}" at index=2
```